esp-box-lite/esp-box-lite.c

```diff
@@ -415,3 +420,30 @@ void bsp_display_unlock(void)
 {
     lvgl_port_unlock();
 }
+
+esp_err_t bsp_iot_button_create(button_handle_t btn_array[], int *btn_cnt, int btn_array_size)
+{
+    esp_err_t ret = ESP_OK;
+    if ((btn_array_size < BSP_BUTTON_NUM) ||
```

Comment:
The array `btn_array_size` can be smaller (without BSP_BUTTON_PREV, BSP_BUTTON_ENTER, BSP_BUTTON_NEXT buttons)

Reply:
> The array `btn_array_size` can be smaller (without BSP_BUTTON_PREV, BSP_BUTTON_ENTER, BSP_BUTTON_NEXT buttons)

Yes，I have the same consideration, this operation wastes three useless handles.
But I want to maintain the consistency of the interface.
I didn't think of a particularly suitable method, do you have any suggestion here?

Reply:
If I am thinking about it, we can remove navigation buttons from `bsp_button_t` and from `bsp_button_config`. We can left only public buttons for user usage and this three buttons will be private onsice `.c` file.
What do you think?

Reply:
> If I am thinking about it, we can remove navigation buttons from `bsp_button_t` and from `bsp_button_config`. We can left only public buttons for user usage and this three buttons will be private onsice `.c` file. What do you think?

Yes,  it needs to update `esp_lv_port`. We update this time, or the next step?

Reply:
Why need update `esp_lvgl_port`? I thought only move these three buttons into `esp-box-lite.c` and use it same.

Reply:
> Why need update `esp_lvgl_port`? I thought only move these three buttons into `esp-box-lite.c` and use it same.


OK，I understand what you mean, just define a private strcut for ADC button here.
I thought you wanted to take out the initialization of the `navigation buttons` in `esp_lv_port` and pass them the handle.

Reply:
So the user won't be able to user the navigation buttons without LVGL, right? Hmmm, seems like a breaking change to me...

Are you sure this is the best option?

Reply:
Before user couldn't use it too, because it was used for navigation. It was only in the same structure. I think, this is better.

Reply:
OK